src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -481,11 +481,12 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 		this.renderContentReferencesIfNeeded(element, templateData, templateData.elementDisposables);
 
 		let fileTreeIndex = 0;
+		let codeBlockIndex = 0;
 		value.forEach((data, index) => {
 			const result = data.kind === 'treeData'
 				? this.renderTreeData(data.treeData, element, templateData, fileTreeIndex++)
 				: data.kind === 'markdownContent'
-					? this.renderMarkdown(data.content, element, templateData, fillInIncompleteTokens)
+					? this.renderMarkdown(data.content, element, templateData, fillInIncompleteTokens, codeBlockIndex++)
```

Comment:
Doesn't this need a fix in `doNextProgressiveRender` too?

Reply:
Tested in main, this is not a regression FYI. I don't know if you were planning on taking it as a candidate, if it is easier to hit this in `@vscode` now than before, we could still do a fix like this where it will look wrong as it streams but get fixed up when the response is done loading. Gotta think about the right way to track the codeblock offsets for progressive rendering.

Reply:
1. I do want to take it in as a candidate. Updated label and milestone. 
2.  > Doesn't this need a fix in doNextProgressiveRender too
     
     We do need a fix for 👆.  I already have an issue for this, but the fix is not worth the risk now. The behavior without the fix is not wrong. It's just glitchy as shown below. It's only wrong with the code block references when re-rendering after the response is done loading which this PR should fix for now. 
     
https://github.com/microsoft/vscode/assets/25044782/cd86c81b-e914-45cf-a3af-d2293d1982cb